pkg/server/plugin/keymanager/awskms/awskms_test.go

```diff
@@ -0,0 +1,708 @@
+		aliases                []*kms.AliasListEntry
+		keyType                keymanager.KeyType
+		keySpec                string
+		publicKey              string
+		createKeyErr           string
+		getPublicKeyErr        string
+		scheduleKeyDeletionErr string
+	}{
+		{
+			name: "non existing key",
+			aliases: []*kms.AliasListEntry{
+				{
+					AliasName:   aws.String(spireKeyAlias),
+					TargetKeyId: aws.String(kmsKeyID),
+				},
+			},
+			expectedEntries: map[string]keyEntry{
+				spireKeyID: {
+					KMSKeyID: kmsKeyID,
+					PublicKey: &keymanager.PublicKey{
+						Id:   spireKeyID,
+						Type: keymanager.KeyType_RSA_4096,
+					},
+				},
+			},
+		},
+		{
+			name: "replace old key",
+			aliases: []*kms.AliasListEntry{
+				{
+					AliasName:   aws.String(spireKeyAlias),
+					TargetKeyId: aws.String(kmsKeyID),
+				},
+			},
+			expectedEntries: map[string]keyEntry{
+				spireKeyID: {
+					KMSKeyID: kmsKeyID,
+					PublicKey: &keymanager.PublicKey{
+						Id:   spireKeyID,
+						Type: keymanager.KeyType_RSA_4096,
+					},
+				},
+			},
+		},
+		{
+			name:    "unsupported key spec",
+			keyType: keymanager.KeyType_RSA_1024,
+			err:     "awskms: unsupported key type: KeyType_RSA_1024",
+			aliases: []*kms.AliasListEntry{
+				{
+					AliasName:   aws.String(spireKeyAlias),
+					TargetKeyId: aws.String(kmsKeyID),
+				},
+			},
+		},
+		{
+			name:         "create key error",
+			err:          "awskms: failed to create key: fake key",
+			createKeyErr: "fake key",
+			aliases: []*kms.AliasListEntry{
+				{
+					AliasName:   aws.String(spireKeyAlias),
+					TargetKeyId: aws.String(kmsKeyID),
+				},
+			},
+		},
+		{
+			name:            "get public key error",
+			err:             "awskms: failed to get public key: public key error",
+			getPublicKeyErr: "public key error",
+			aliases: []*kms.AliasListEntry{
+				{
+					AliasName:   aws.String(spireKeyAlias),
+					TargetKeyId: aws.String(kmsKeyID),
+				},
+			},
+		},
+		{
+			name:                   "schedule key deletion error",
+			scheduleKeyDeletionErr: "schedule key deletion error",
+			aliases: []*kms.AliasListEntry{
+				{
+					AliasName:   aws.String(spireKeyAlias),
+					TargetKeyId: aws.String(kmsKeyID),
+				},
+			},
+			expectedEntries: map[string]keyEntry{
+				spireKeyID: {
+					KMSKeyID: kmsKeyID,
+					PublicKey: &keymanager.PublicKey{
+						Id:   spireKeyID,
+						Type: keymanager.KeyType_RSA_4096,
+					},
+				},
+			},
+		},
+	} {
+		tt := tt
+		t := ps.T()
+		t.Run(tt.name, func(t *testing.T) {
+			ps.reset()
+			ps.setupScheduleKeyDeletion("")
+			ps.setupListAliases(tt.aliases, "")
+			ps.setupDescribeKey(kms.CustomerMasterKeySpecRsa4096, "")
+			ps.setupCreateKey(kms.CustomerMasterKeySpecRsa4096, tt.createKeyErr)
+			ps.setupGetPublicKey("")
+
+			_, err := ps.plugin.Configure(ctx, ps.configureRequestWithDefaults())
+			ps.Require().NoError(err)
+
+			ps.setupGetPublicKey(tt.getPublicKeyErr)
+
+			keyType := keymanager.KeyType_RSA_4096
```

Comment:
nit: declare this as a const before the test templates and reference it in the respective templates to make it a little more clear why the expected results in some of the test cases should have this key type

Reply:
I will address this one on the next commit. Where I plan to increase the coverage and some changes.